lib/wikidata/diff/api.rb

```diff
@@ -22,15 +22,52 @@ def self.get_revision_contents(revision_ids)
     raise e
   end
 
-  def self.fetch_revision_data(client, revision_ids)
-    client.action(
-      'query',
+  def self.fetch_revision_data(revision_ids)
+    query_parameters = {
       prop: 'revisions',
       revids: revision_ids&.join('|'),
       rvslots: 'main',
       rvprop: 'content|ids|comment',
       format: 'json'
-    )
+    }
+    fetch_all_revisions(query_parameters)
+  end
+
+  def self.fetch_all_revisions(query)
+    client = api_client
+    data = {}
+    continue_param = nil
+
+    loop do
+      query.merge!(continue_param) if continue_param
+      response = mediawiki_request(client, 'query', query)
+      break unless response
+
+      data.deep_merge!(response.data)
+      continue_param = response['continue']
+      break if continue_param.nil?
+    end
+
+    data
+  end
+
+  def self.mediawiki_request(client, action, query)
+    tries ||= 3
+    client.send(action, query)
+  rescue StandardError => e
+    tries -= 1
+    sleep 1 if too_many_requests?(e)
+    retry unless tries.zero?
+    log_error(e)
```

Comment:
Where does this method come from?

Reply:
I saw it here: https://github.com/WikiEducationFoundation/WikiEduDashboard/blob/d5ec317a43ea1f2bd43c34b71a2851717cc40b7a/lib/wiki_api.rb#L100

Reply:
That function comes from Dashboard code, and shouldn't be used in this gem.

I think raising the error and letting the Dashboard code handle it is fine here.

Reply:
Okay, will work on that

Reply:
That case, you don't even need the rescue. However, I think the retry behavior is worth keeping. The idea would just be to raise the error instead of logging it, if the retries are exhausted.